lib-asttoir/src/lib.rs

```diff
@@ -0,0 +1,238 @@
+#[macro_use]
+extern crate lazy_static;
+
+use ir;
+use std::collections::HashMap;
+use std::sync::Mutex;
+
+lazy_static! {
+    static ref IS_EXPRESSION: HashMap<&'static str, bool> = {
+        let mut map = HashMap::new();
+        map.insert("ThisExpression", true);
+        map.insert("Identifier", true);
+        map.insert("Literal", true);
+        map.insert("ArrayExpression", true);
+        map.insert("ObjectExpression", true);
+        map.insert("FunctionExpression", true);
+        map.insert("ArrowFunctionExpression", true);
+        map.insert("TaggedTemplateExpression", true);
+        map.insert("MemberExpression", true);
+        map.insert("TaggedTemplateExpression", true);
+        map.insert("Super", true);
+        map.insert("MetaProperty", true);
+        map.insert("NewExpression", true);
+        map.insert("TaggedTemplateExpression", true);
+        map.insert("CallExpression", true);
+        map.insert("UpdateExpression", true);
+        map.insert("AwaitExpression", true);
+        map.insert("UnaryExpression", true);
+        map.insert("BinaryExpression", true);
+        map.insert("ConditionalExpression", true);
+        map.insert("YieldExpression", true);
+        map.insert("SequenceExpression", true);
+
+        map
+    };
+}
```

Comment:
Hmm, I'm wondering if the performance of a simple match is bad enough that you need to use a HashMap like this.

```rust
fn is_expression(typ: &str) -> bool {
    match typ {
        "ThisExpression" => true,
        "Identifier" => true,
        "Literal" => true,
        "ArrayExpression" => true,
        "ObjectExpression" => true,
        "FunctionExpression" => true,
        "ArrowFunctionExpression" => true,
        "TaggedTemplateExpression" => true,
        "MemberExpression" => true,
        "TaggedTemplateExpression" => true,
        "Super" => true,
        "MetaProperty" => true,
        "NewExpression" => true,
        "TaggedTemplateExpression" => true,
        "CallExpression" => true,
        "UpdateExpression" => true,
        "AwaitExpression" => true,
        "UnaryExpression" => true,
        "BinaryExpression" => true,
        "ConditionalExpression" => true,
        "YieldExpression" => true,
        "SequenceExpression" => true,
        _ => false
    }
}
```